src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -59,9 +59,24 @@ IList<I> GetReferencedItems(string itemType, ImmutableHashSet<string> globsToIgn
                 }
             }
 
-            public abstract void Apply(ImmutableList<ItemData>.Builder listBuilder, ImmutableHashSet<string> globsToIgnore);
+            public virtual void Apply(ImmutableList<ItemData>.Builder listBuilder, ImmutableHashSet<string> globsToIgnore)
+            {
+                var items = SelectItems(listBuilder, globsToIgnore).ToList();
+                MutateItems(items);
```

Comment:
I wrote a unit test and it seems to work:

```xml
                            <i2 Include='a;b;c'>
                                <m1>m1_contents</m1>
                                <m2>m2_contents</m2>
                            </i2>

                            <i Include='@(i2)'/>

                            <i2 Update='a;b;c'>
                                <m1>m1_updated</m1>
                                <m2>m2_updated</m2>
                            </i2>
```

Do you have another case in mind?